### ISSUE TITLE
Bug: ERROR validating provider specific settings: IPv4 key is not set

### ISSUE THREAD

**Issue by thany** [qdm12/ddns-updater#875]
**TLDR**: Custom updater shouldn't require IPv4 key

1. Is this urgent: Yes
2. DNS provider(s) you use: Nederhost (custom)
3. Program version:

    

    `Running version latest built on 2024-11-21T22:27:40.202Z (commit 89564b7)`

4. What are you using to run the container: docker
5. Extra information (optional)
    
    The service I'm using doesn't require passing the IP address. It just uses the IP address that's calling the service. So all functionality to get the current public IP adress can be disabled when the IPv4 key is not present or empty.
    
    I've marked this as urgent because the whole ddns-updater service refuses to start. Now, I *could* probably add a bogus key and call it a day, but I will 100% forget about it, plus I shouldn't "litter" by sending `foo=0.0.0.0` that's only there because an updater can't handle not sending this bit.

Configuration file:

```json
  "settings": [
    {
      "provider": "custom",
      "domain": "example.org",
      "url": "https://****:****@dyndns.nederhost.nl/nic/update/?hostname=example.org",
      "success_regex": "good|nochg",
      "ipv4key": "",
      "ipv6key": ""
    }
  ]
```

Host OS: Unraid

**Comment by qdm12** [comment]
You can now set ip keys to the empty string and the program will just not send a query parameter. This is available in the latest image and in the future v2.9.0 release.
